nixos/modules/services/web-apps/photoprism.nix

```diff
@@ -0,0 +1,291 @@
+        <literal>"mysql"</literal> driver) to improve performance.
+      '';
+    };
+
+    databaseDsn = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = ''
+        photoprism@unix(/run/mysqld/mysqld.sock)/photoprism?charset=utf8mb4,utf8&parseTime=true
+      '';
+      description = ''
+        Photoprism database source name. Valid only when
+        <link linkend="opt-services.photoprism.databaseDriver">databaseDriver</link>
+        is set to <literal>"mysql"</literal>.
+      '';
+    };
+
+    initialPassword = mkOption {
+      type = types.str;
+      default = "insecure";
+      description = ''
+        Initial admin password.
+
+        This should be changed in the web UI after the first startup.
+      '';
+    };
+
+    detectNsfw = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Flag photos as private that MAY be offensive.
+      '';
+    };
+
+    allowNsfw = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Allow uploads that MAY be offensive.
+      '';
+    };
+
+    experimental = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable experimental features.
+      '';
+    };
+
+    originalsLimit = mkOption {
+      type = types.int;
+      default = 5000;
+      description = ''
+        File size limit for originals in MB (increase for high-res video).
+      '';
+    };
+
+    public = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        No authentication required (disables password protection).
+      '';
+    };
+
+    readonly = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Do not modify originals directory (reduced functionality).
+      '';
+    };
+
+    originalsDir = mkOption {
+      type = types.path;
+      example = "/mnt/nas/photos";
+      description = "Path to your photos.";
+    };
+
+    dataDir = mkOption {
+      type = types.path;
+      default = "/var/lib/photoprism";
+      description = "Data directory for photoprism.";
+    };
+
+    siteTitle = mkOption {
+      type = types.str;
+      default = "PhotoPrism";
+      description = "Web UI title.";
+    };
+
+    siteCaption = mkOption {
+      type = types.str;
+      default = "Browse Your Life";
+      description = "Web UI caption.";
+    };
+
+    extraEnv = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      example = { PHOTOPRISM_DEBUG = "true"; };
+      description = ''
+        Additional environment variables for the photoprism service.
+
+        See <link xlink:href="https://dl.photoprism.app/docker/docker-compose.yml"/>
+        and <link xlink:href="https://github.com/photoprism/photoprism/blob/develop/internal/config/flags.go"/>
+        for the supported environment variables.
+      '';
+    };
+
+    package = mkOption {
```

Comment:
Why is this option needed?

Reply:
I can't see a reason not to.

It's just a couple of lines, allows for more flexibility, and exists in most services modules.
![image](https://user-images.githubusercontent.com/5727578/172502317-ced04bb8-6758-4cb1-95ea-c3f5558f9561.png)

Reply:
Let's remove the option until someone finds a need for it.